src/ImageSharp/Memory/MemoryAllocator.cs

```diff
@@ -9,23 +9,22 @@ namespace SixLabors.Memory
     public abstract class MemoryAllocator
     {
         /// <summary>
-        /// Allocates an <see cref="IBuffer{T}"/> of size <paramref name="length"/>, optionally
-        /// clearing the buffer before it gets returned.
+        /// Allocates an <see cref="IBuffer{T}"/> of size <paramref name="length"/>.
         /// </summary>
         /// <typeparam name="T">Type of the data stored in the buffer</typeparam>
         /// <param name="length">Size of the buffer to allocate</param>
-        /// <param name="clear">True to clear the backing memory of the buffer</param>
+        /// <param name="options">The allocation options.</param>
         /// <returns>A buffer of values of type <typeparamref name="T"/>.</returns>
-        internal abstract IBuffer<T> Allocate<T>(int length, bool clear)
+        internal abstract IBuffer<T> Allocate<T>(int length, AllocationOptions options = AllocationOptions.None)
```

Comment:
I know there were arguments against using parameter defaults, but on places like this it feels cleaner to me. [ArrayPool.Return](https://github.com/dotnet/corefx/blob/master/src/System.Buffers/ref/System.Buffers.cs#L16) looks basically the same.

@JimBobSquarePants thoughts?

Reply:
Overload?